dev/test/specs/starter-test.js

```diff
@@ -231,8 +248,28 @@ describe('test basic arcs functionality', function() {
     acceptSuggestion(footerPath, 'make a reservation');
     acceptSuggestion(footerPath, 'You are free');
 
+    browser.close();
     // to drop into debug mode with a REPL; also a handy way to see the state
     // at the end of the test:
     // browser.debug();
   });
+
+  it('can use the gift shopping demo flow', function() {
+    initTestWithNewArc();
+
+    const footerPath = getFooterPath();
+    allSuggestions(footerPath);
+
+    acceptSuggestion(footerPath, "Show Products from your browsing context (Minecraft Book plus 2 other items) and choose from Products recommended based on Products from your browsing context and Claire's wishlist (Book: How to Draw plus 2 other items)");
+
+    // TODO: click the "Add" buttons to move products from recommended to shortlist and
+    // (1) verify product was moved,
+    // (2) verify 'action' slot is not visible after all products were moved.
+
+    acceptSuggestion(footerPath, "Estimate arrival dates, estimate arrival dates");  // TODO: add "and buy gifts for Claire" when descriptions are fixed.
+    acceptSuggestion(footerPath, "check manufacturer information for Products from your browsing context");
+    acceptSuggestion(footerPath, "recommendations based on Products recommended based on Products from your browsing context and Claire's wishlist");
```

Comment:
Is there an assertion that'd make sense at the end, so we now this last suggestion worked?

Reply:
added some asserts, will add more later (there is a TODO), but i need additional helper methods first.